csrc/multidevice/utils.cpp

```diff
@@ -53,9 +29,20 @@ std::vector<IterDomain*> getShardedIterDomains(TensorView* tv) {
       [](auto id) { return id->isDeviceDim(); });
   return sharded_ids;
 }
-
 } // namespace
 
+bool isSharded(TensorView* tv) {
+  auto sharded_domains = getShardedIterDomains(tv);
```

Comment:
All of the multidevice tests passed, but we don't have any tests that generate a Reduction+Device axes. i.e. we don't have DIDx(i0) -> r(DIDx(i0)). We have discussed how that this is valid. 
I'll update isSharded so that we ignore reduction domains in the analysis.

Reply:
My mistake, we do test this! There are just in a different test executable than test_multidevice. https://github.com/NVIDIA/Fuser/blob/cccfa1fb578022ea6a3866c30256cad5071a03ca/test/test_pipeline.cpp#L85

Reply:
oh right. We should probably move everything to the same test file...

Reply:
I have some test refactors for a later PR. I can open a PR for test cleanup.